server/models/User.coffee

```diff
@@ -264,12 +264,7 @@ UserSchema.statics.unconflictName = unconflictName = (name, done) ->
 
 UserSchema.methods.register = (done) ->
   @set('anonymous', false)
-  if (name = @get 'name')? and name isnt ''
-    unconflictName name, (err, uniqueName) =>
-      return done err if err
-      @set 'name', uniqueName
-      done()
-  else done()
+  done()
```

Comment:
This function doesn't ever need to be asynchronous anymore. The function that calls it can just call it normally, now, without a callback to wait for.